### PR TITLE
dot/rpc: implement chain_subscribeFinalizedHeads RPC method

### LINKED ISSUE #1139: polkadot.js api not receiving messages from chain.subscribeFinalizedHeads
## Expected Behavior

- polkadot.js api rpc.chain.subscribeFinalizedHeads should get message when a block is finalized.

## Current Behavior

- No response is received from polkadot.js api rpc.chain.subscribeFinalizedHeads when a block is finalized.

## Possible Solution

- Related to #790 

## Steps to Reproduce

1. example call:
```
...
 await api.rpc.chain.subscribeFinalizedHeads((lastFin) => {
                console.log(`${chain}: last finalized block #${lastFin.number} has hash ${lastFin.hash}`)       
});
...
```
2.
3.
4.

## Specification

- go version:
- gossamer version:
- gossamer commit tag:
- gossamer commit hash:
- operating system:
- additional links:

## Checklist

- [x] I have read [CODE_OF_CONDUCT](https://github.com/ChainSafe/gossamer/blob/development/.github/CODE_OF_CONDUCT.md) and [CONTRIBUTING](https://github.com/ChainSafe/gossamer/blob/development/.github/CONTRIBUTING.md) 
- [x] I have provided as much information as possible and necessary
- [x] I am planning to submit a pull request to fix this issue myself

### PR DESCRIPTION
**Pull request by edwardmack**
## Changes

- Implement chain_subscribeFinalizedHeads RPC method which is used by polkadot.js.
-
-

## Tests

```
go test ./dot/rpc/...
```

## Checklist

- [x] I have read [CODE_OF_CONDUCT](https://github.com/ChainSafe/gossamer/blob/development/.github/CODE_OF_CONDUCT.md) and [CONTRIBUTING](https://github.com/ChainSafe/gossamer/blob/development/.github/CONTRIBUTING.md) 
- [x] I have provided as much information as possible and necessary
- [x] I have reviewed my own pull request before requesting a review
- [x] All integration tests and required coverage checks are passing

## Issues

- fixes #1139

### CODE REVIEW

**Review by noot — APPROVED** [review]
looks good, please add a test!

File: tests/utils/gossamer_utils.go [hunk 1]
```diff
@@ -108,43 +112,38 @@ func InitGossamer(idx int, basePath, genesis, config string) (*Node, error) {
...
 // StartGossamer starts given node
-func StartGossamer(t *testing.T, node *Node) error {
+func StartGossamer(t *testing.T, node *Node, websocket bool) error {
 	var key string
+	var params []string = []string{"--port", strconv.Itoa(basePort + node.Idx),
+		"--config", node.config,
+		"--basepath", node.basePath,
+		"--rpchost", HOSTNAME,
+		"--rpcport", node.RPCPort,
+		"--rpcmods", "system,author,chain,state",
```
Comment (noot) [inline tests/utils/gossamer_utils.go:129]: add `dev` here?

File: tests/rpc/rpc_03-chain_test.go [hunk 2]
```diff
@@ -135,3 +137,162 @@ func TestChainRPC(t *testing.T) {
...
+
+	testCases := []*testCase{
+		{
+			description: "test chain_subscribeNewHeads",
+			method:      "chain_subscribeNewHeads",
+			expected: []interface{}{1,
+				map[string](interface{}){
+					"subscription": float64(1),
+					"result": map[string](interface{}){
+						"number":         "0x04",
```
Comment (noot) [inline tests/rpc/rpc_03-chain_test.go:155]: is there a reason the expected block is block 4?

Comment (edwardmack) [reply tests/rpc/rpc_03-chain_test.go:155]: No, 1 makes more sense, I've updated this.

**Review by noot — APPROVED** [review]

**Pull request merged**